### PR TITLE
Leverage Daft Transient Errors to retry tasks on ray and drop timeout times

### PR DESCRIPTION
**Pull request by samster25**
* Upgrade Daft to 0.2.23
* Cut down connect timeout to 5 seconds and read timeout to 10 seconds
* Add Daft Transient error to retry list for ray
* This version of daft also will not panic on connection issues.

### CODE REVIEW

**Comment by samster25** [comment]
This leverages: https://github.com/Eventual-Inc/Daft/pull/2197 
and https://github.com/Eventual-Inc/Daft/pull/2214

**Review by raghumdani — COMMENTED** [review]
Thanks for this PR. Are connections per file still relevant?

**Comment by raghumdani** [comment]
We are upgrading Ray version. We will plan this deployment after observing new Ray version to isolate issues that may be related to either of the changes.

File: deltacat/utils/daft.py [hunk 1]
```diff
@@ -163,5 +163,7 @@ def _get_s3_io_config(s3_client_kwargs) -> IOConfig:
             retry_mode="adaptive",
             num_tries=BOTO_MAX_RETRIES,
             max_connections=DAFT_MAX_S3_CONNECTIONS_PER_FILE,
+            connect_timeout_ms=5_000,  # Timeout to connect to server
```
[the remark this answers is not in the document]
Comment (samster25) [reply deltacat/utils/daft.py:166]: If we want to do that, we should be able to retry on `DaftTransientError` for tenacity.

[the remark this answers is not in the document]
Comment (raghumdani) [reply deltacat/utils/daft.py:166]: Will take up a separate PR.

**Review by raghumdani — APPROVED** [review]
LGTM! Thanks for this change.

**Pull request merged**